### PR TITLE
[Refactor] CSL

### PR DESCRIPTION
**Pull request by zytx121**
- [x] Align test
- [x] Align train
- [x] UT

## Experiments
rotated_retinanet_obb_csl_gaussian_r50_fpn_fp16_1x_dota_le90.py 
report AP50: 69.51
AP50: 0.6804708695155224
AP75: 0.39561891292657414
mAP: 0.3920965305629459

### CODE REVIEW

File: mmrotate/core/bbox/coder/angle_coder.py [hunk 1]
```diff
@@ -37,16 +38,16 @@ def __init__(self, angle_version, omega=1, window='gaussian', radius=6):
         self.radius = radius
         self.coding_len = int(self.angle_range // omega)
```
Comment (liuyanyi) [inline mmrotate/core/bbox/coder/angle_coder.py:39]: Is that better to use `encoded_size` to be consistent with `BaseBBoxCoder`?

File: mmrotate/core/bbox/coder/angle_coder.py [hunk 2]
```diff
@@ -105,7 +106,7 @@ def decode(self, angle_preds):
                 Has shape (num_anchors * H * W, coding_len)
 
         Returns:
-            list[Tensor]: Angle offset for each scale level.
+            Tensor: Angle offset for each scale level.
                 Has shape (num_anchors * H * W, 1)
         """
         angle_cls_inds = torch.argmax(angle_preds, dim=1)
```
Comment (liuyanyi) [inline mmrotate/core/bbox/coder/angle_coder.py:112]: The shape of returns is wrong, i think the return here is (num_anchors * H * W).
In order to support batch input like (B, N, coding_len), the argmax dim can be -1.


In retinanet, csl directly assign the angle into rbox
```
# `rbbox` has shape (N, 5)
# `ang` has shape (N)
rbbox[..., 4] = ang
```
In some angle_branch head, such as Rotated FCOS and Rotated YOLOX, they use concat to get final bbox.
```
# `hbbox` has shape (N, 4)
# `ang` has shape (N, 1)
rbbox = torch.cat([hbbox, ang], dim=-1)
```

I think we can add a `keepdim` arg in decode method, and use like this:
```
angle_cls_inds = torch.argmax(angle_preds, dim=-1, keepdim=keepdim)
```
When keepdim is true, return (N, 1) or (B, N, 1), otherwise (N,) or (B, N)

Comment (zytx121) [reply mmrotate/core/bbox/coder/angle_coder.py]: Done.

**Review by liuyanyi — APPROVED** [review]

**Review by RangiLyu — APPROVED** [review]
